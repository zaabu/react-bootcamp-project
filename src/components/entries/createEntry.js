import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import CreateEntryView   from "../../views/entries/createEntry";
import { createDiaryEntry } from "../../actions/entryActions/entryAction";


class CreateDiaryEntryComponent extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            modal: false,
            title: "",
            body: ""
        };
    }

    toggle = () => {
        this.setState({
            modal: !this.state.modal
        });
    };

    handleChange = event => {
        this.setState({
            [event.target.name]: event.target.value
        });
    };

    handleSubmit = () => {
        const data = {
            title: this.state.title,
            body: this.state.body
        };
        this.props.dispatch(createDiaryEntry(data));

    };
    modalClose = () => {
        this.toggle();
        this.handleSubmit();
    }

    render() {
        const { modal, title, body} = this.state;
        const classname = this.props.className;
        return (
            <div>
                <CreateEntryView
                    toggle={this.toggle}
                    modal={modal}
                    title={title}
                    body={body}
                    classname={classname}
                    closemodal={this.modalClose}
                    handleChange={this.handleChange}
                />
            
            </div>
        )
    }
}

CreateDiaryEntryComponent.propTypes = {
    className: PropTypes.string,
    dispatch: PropTypes.func
};

const mapDispatchToProps = dispatch => ({ dispatch });
const mapStateToProps = state => ({
    entry: state.entriesReducer

});

export {CreateDiaryEntryComponent as CreateEntryModalTest};
export default connect(
    mapStateToProps,
    mapDispatchToProps

)(CreateDiaryEntryComponent);