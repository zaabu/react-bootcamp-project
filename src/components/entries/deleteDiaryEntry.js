import React, { Component } from "react";
import PropTypes from "prop-types";
import { confirmAlert } from "react-confirm-alert";
import "react-confirm-alert/src/react-confirm-alert.css";


class DeleteEntryConfirm extends Component {
    submit = (id) => {
        confirmAlert({
            title: "Delete Entry",
            message: "Are you sure you want to delete your entry.",
            buttons: [
                {
                    label: "Yep",
                    onClick: () => this.props.delete(this.props.uniqueId)
                },
                {
                    label: "Nope"
                }
            ]
        });
    };

    render() {
        return <button id="danger-button" type="button" class="btn btn-danger"  onClick={this.submit} href="#">Delete</button>
    }

    
}
DeleteEntryConfirm.propTypes = {
    handleDelete: PropTypes.func,
    uniqueId: PropTypes.id
        
}

export default DeleteEntryConfirm;