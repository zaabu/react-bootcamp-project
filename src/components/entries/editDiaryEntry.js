import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import EditEntryView from "../../views/entries/editEntryView";
import { updateDiaryEntry } from "../../actions/entryActions/entryAction";


class EditEntryModal extends React.Component {
    constructor(props) {
      super(props);
      this.state = {
        modal: false,
        title: this.props.title,
        body: this.props.body
      };
    }
  
    toggle = () => {
      this.setState({
        modal: !this.state.modal
      });
    };
  
    handleChange = event => {
      this.setState({
        [event.target.name]: event.target.value
      });
    };
  
    handleUpdate = () => {
      const data = {
        
        title: this.state.title,
        body: this.state.body,
        // type: this.state.type
      };
      this.props.dispatch(updateDiaryEntry(this.props.uniqueId,data));
    };
    modalClose = () => {
      this.toggle();
      this.handleUpdate();
    };
  
    render() {
      const { modal,title, body } = this.state;
      const classname = this.props.className;
      return (
        <React.Fragment>
          <EditEntryView
            toggle={this.toggle}
            modal={modal}
            title={title}
            body={body}
            classname={classname}
            closemodal={this.modalClose}
            handleChange={this.handleChange}
            props={this.props}
          />
        </React.Fragment>
      );
    }
  }
  EditEntryModal.propTypes = {
    className: PropTypes.string,
    dispatch: PropTypes.func,
    props: PropTypes.object,
    uniqueId:PropTypes.string
  };
  const mapDispatchToProps = dispatch => ({ dispatch });
  const mapStateToProps = state => ({
    modified : state.entriesReducer.modified
  });
//   export {EditEntryModal as EditEntryModalTest};
  export default connect(
    mapStateToProps,
    mapDispatchToProps
  )(EditEntryModal);
