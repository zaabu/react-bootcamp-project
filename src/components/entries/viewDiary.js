import React from "react";
import PropTypes from "prop-types";


import {
  Modal,
  ModalHeader,
  ModalBody,
  ModalFooter,
  Badge
} from "reactstrap";


 class ViewEntry extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      modal: false
    };
     this.toggle = this.toggle.bind(this);
  }
   toggle() {
    this.setState({
      modal: !this.state.modal
    });
  }

  render() {
    const closeBtn = (
      <button className="close" onClick={this.toggle}>
        &times;
      </button>
    );
    return (
      <React.Fragment>
        <button type="button" id="success-button" class="btn btn-success" onClick={this.toggle} >View</button>
        <Modal
          isOpen={this.state.modal}
          toggle={this.toggle}
          className={this.props.className}
        >
          <ModalHeader toggle={this.toggle} close={closeBtn}>
            {this.props.title}
          </ModalHeader>
          <ModalBody>{this.props.body}</ModalBody>
          <ModalFooter>
            <Badge  className="pull-right float-left">
              Date Created - {this.props.date_created}
            </Badge>{" "}
            <Badge  className="pull-right float-left">
              Last Modified - {this.props.date_modified}
            </Badge>
          </ModalFooter>
        </Modal>
      </React.Fragment>
    );
  }
}
ViewEntry.propTypes = {
  buttonLabel: PropTypes.string,
  className: PropTypes.string,
  title: PropTypes.string,
  body: PropTypes.string,
  date_modified: PropTypes.string,
  date_created: PropTypes.string
};
 export default ViewEntry;