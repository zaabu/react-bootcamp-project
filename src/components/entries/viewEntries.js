import React, { Component } from "react";
import Viewallentries from "../../views/entries/ViewEntries";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import { retrieveAllEntries, deleteDiaryEntry } from "../../actions/entryActions/entryAction";
import CreateDiaryEntryComponent from "../../components/entries/createEntry";
// import DeleteEntryConfirm from "../../components/entries/deleteEntry";



export class ViewEntries extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  componentWillReceiveProps(nextProps) {
    if (
      nextProps.entry.message === "diary entry added successfully." ||
      nextProps.deleted.message === "Diary entry deleted successfully" ||
      nextProps.modified.message === "diary entry updated succesfully"
    ){
      this.props.dispatch(retrieveAllEntries())
    }
  }

  handleDelete = id => {
    this.props.dispatch(deleteDiaryEntry(id));
  }

  componentWillMount() {
    this.props.dispatch(retrieveAllEntries());
  }

  render() {
    const entries = this.props.allentries;
    
    return (
      <div>
        {/* <h1 class="title d-inline">MY DIARY</h1> */}
       
        
        <CreateDiaryEntryComponent />
        {/* <div class="button_cont d-inline user-button example_a" > */}
        {/* <div class="button_cont d-inline" ><a class="example_a" href="add-website-here" target="_blank" rel="nofollow noopener">Log Out</a></div> */}
        
        {/* <h2 class="d-inline pencil-icon"><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAADIAAAAyCAYAAAAeP4ixAAAAAXNSR0IArs4c6QAAAARnQU1BAACxjwv8YQUAAAAJcEhZcwAADsMAAA7DAcdvqGQAAAGLSURBVGhD7dq7SgNRGEXhUUGtFAuvvYKKWImthfgkIpjW0kpsBCvfQMRKFPFeamVlYeOttlBB0EYELXTtwIFhmInOJJL/xLPhKxJSnJVAEjKJwsJyrRnzuMQnXrGFUXizVmzjK8UbpuHFRqBXIC1EntEJLzaBSjGzMLl27OAe7tmuFLMGc1PECXTAB3TALStmBaYWj3jEMJJLi5mBmbXhpwi3eMwpmmBieSLcFHOO7vItA8sT0YUlDJZvGVreiAvosau6w8oUcYy8EXfoh4lVEzEAEysacYsQUcvpW2yIsLB/H2HmLVYRRwgR9V7RiBuEiFquYSIO4SL0y0fWQsRfrpqIPpiYIg4QIuq9ohHXCBG1XMNE7CNE1HvVRPTCxBSxB68jNHe5y+uIIehgHxjTHRmLR1zBVIRWgg4nG2hBcuYjtF24kLQYLyJ04BfEQ+IxXkRok0hGOJvwIkJbRDIgSRE9ML0zpB1enrAO8xHaO9zB9RcJXVFawDjMXHT8zeawjCno0z0srPCi6BsHNBINm/BYCQAAAABJRU5ErkJggg=="></img></h2> */}

        <Viewallentries results={entries}  handleDelete={this.handleDelete} />
      </div>
    );
  }
}
ViewEntries.propTypes = {
  dispatch: PropTypes.func.isRequired,
  allentries: PropTypes.array,
  entry: PropTypes.object,
  deleted: PropTypes.object,
  message: PropTypes.string,
  modified: PropTypes.object


};


const mapStateToProps = state => ({
  allentries: state.entriesReducer.entries,
  entry: state.entriesReducer.entry,
  deleted: state.entriesReducer.deleted,
  modified: state.entriesReducer.modified
  
});
const mapDispatchToProps = dispatch => ({ dispatch });

export { ViewEntries as ViewDiariesTest }
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ViewEntries);
