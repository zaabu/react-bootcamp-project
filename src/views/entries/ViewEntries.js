import React from "react";
import "../../assets/entryAssets/entrypage.scss";
import PropTypes from "prop-types";
import DeleteEntryConfirm from "../../components/entries/deleteDiaryEntry";
import EditEntryModal from "../../components/entries/editDiaryEntry"
import ViewEntry from "../../components/entries/viewDiary";


const Viewallentries = ({ results, handleDelete }) => {
  const showEntries = results.map(entry => (
    entriesView(entry, handleDelete)
  ));

  return (
    <div>
      <br />
      <div className="pic-container">{showEntries}</div>
    </div>
  );
};
Viewallentries.propTypes = {
  results: PropTypes.array,
};
export default Viewallentries;


function entriesView(entry, handleDelete) {
  return <div className="article-column" key={entry.id}>
    {/* {entry.title} {" "} */}
    <div className="entry-info">
      <br />
      <header className="entry-header">
        <h3 className="entry-title">
          {entry.title}
        </h3>
      </header>
      <div className="entry-meta">
        <h5 className="entry-description">
          <span>{entry.body.substring(0, 25)}...</span>
          <br />
        </h5>
        <br />
      </div>

      <div class="buttons">
        <div>
          <time class="d-inline float-left entry-date">{entry.date_created}</time>
        </div>

        <div className="d-inline float-right">
          {/* <button class="d-inline view-button">View</button> */}
          {/* <button class="d-inline edit-button">Edit</button> */}
          <ViewEntry 
            title={entry.title}
            body={entry.body}
            date_created = {entry.date_created}
            date_modified = {entry.date_modified}
          />
          <EditEntryModal
            uniqueId={entry.id}
            title={entry.title}
            body={entry.body}
          />
          <DeleteEntryConfirm 
            delete={handleDelete} 
            uniqueId={entry.id} 
          />
        </div>
      </div>

      <br />
    </div>

  </div>;
}

Viewallentries.propTypes = {
  results: PropTypes.array,
  handleDelete: PropTypes.func
};

// export default Viewallentries;

