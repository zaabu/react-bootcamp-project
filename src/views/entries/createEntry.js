import React from "react";
import PropTypes from "prop-types";
import "../../assets/entryAssets/entrypage.scss";
import { clearToken } from "../../utils/myHeaders";
// import "../assets/navbar.scss";

import {
    Button,
    Label,
    Form,
    FormGroup,
    Input,
    Modal,
    ModalHeader,
    ModalBody,
    ModalFooter,
    Collapse,
    Navbar,
    NavbarToggler,
    NavbarBrand,
    Nav,
} from "reactstrap";



const CreateEntryView = ({
    toggle,
    modal,
    classname,
    closemodal,
    handleChange,
    title,
    body,
}) => {
    return (
        <div>
            {/* <a target="_blank" >
                <Button className="user-button" onClick={toggle}>
                    Create
                </Button>
                <Button className="user-button2" href="/" onClick={() => clearToken()}>
                    Log Out
                </Button>
            </a> */}
            <Navbar light expand="md" id="mynav">
                    <NavbarBrand href="/" id="navbar-brand">
                        MY DIARY
                    </NavbarBrand>
                    <NavbarToggler  />
                    <Collapse  navbar >
                    <Nav className="ml-auto" navbar>
                    <ul className="navbar-nav">
                        
                        <a target="_blank" >
                        <Button className="user-button" onClick={toggle}>
                            Create
                        </Button>
                        <Button className="user-button2" href="/" onClick={() => clearToken()}>
                            Log Out
                        </Button>
                        </a>
                        
                        
                    </ul>

                    </Nav>
                    </Collapse>
            </Navbar>
            <Modal isOpen={modal} toggle={toggle} className={classname}>
                <ModalHeader toggle={toggle}>Create New Entry</ModalHeader>
                <ModalBody>
                    <Form>
                        <FormGroup>
                            <Label for="Title">Title</Label>
                            <Input
                             type="text"
                             name="title"
                             id="title"
                             placeholder="Enter entry title.."
                             onChange={handleChange}
                             value={title}
                             required
                            />

                        </FormGroup>
                        <FormGroup>
                            <Label for="Body">Body</Label>
                            <Input
                             type="textarea"
                             name="body"
                             id="body"
                             placeholder="Enter entry body.."
                             onChange={handleChange}
                             value={body}
                             required
                            />

                        </FormGroup>
                    </Form>
                </ModalBody>
                <ModalFooter>
                    <Button className="pull-right" onClick={closemodal} type="submit">
                        Create
                    </Button>
                </ModalFooter>
            
            </Modal>

        </div>
    );
};

CreateEntryView.propTypes = {
    closemodal: PropTypes.func,
    handleChange: PropTypes,
    toggle: PropTypes.func,
    modal: PropTypes.string,
    classname: PropTypes.string,
    title: PropTypes.string,
    body: PropTypes.string
}

export default CreateEntryView;