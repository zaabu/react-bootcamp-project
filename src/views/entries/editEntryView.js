import React from "react";
import PropTypes from "prop-types";
import {
  Button,
  Label,
  Form,
  FormGroup,
  Input,
  Modal,
  ModalHeader,
  ModalBody,
  ModalFooter
} from "reactstrap";
 const EditEntryView = ({
  toggle,
  modal,
  classname,
  closemodal,
  handleChange,
  title,
  body
  
}) => {
  return (
    <React.Fragment>
      {/* <CardLink id="cardlink" onClick={toggle} href="#">
        Edit |
      </CardLink> */}
      {/* <button class="d-inline edit-button" onClick={toggle}>Edit</button>
      <button type="button" class="btn btn-success">Success</button>
      <button type="button" class="btn btn-danger">Danger</button> */}
      <button id="warning-button" type="button" class="btn btn-warning d-inline" onClick={toggle}>Edit</button>
      <Modal isOpen={modal} toggle={toggle} className={classname}>
        <ModalHeader toggle={toggle}>Edit Entry</ModalHeader>
        <ModalBody>
          <Form>
            <FormGroup>
              <Label for="entryTitle">Title</Label>
              <Input
                type="text"
                name="title"
                id="entryTitle"
                placeholder="Enter name of entry ..."
                onChange={handleChange}
                value={title}
                required
              />
            </FormGroup>
            
            <FormGroup>
              <Label for="entryBody">Body</Label>
              <Input
                type="textarea"
                name="body"
                id="entryBody"
                onChange={handleChange}
                value={body}
                required
              />
            </FormGroup>
          </Form>
        </ModalBody>
        <ModalFooter>
          <Button className="pull-right" onClick={closemodal} type="submit">
            Update
          </Button>
        </ModalFooter>
      </Modal>
    </React.Fragment>
  );
};
EditEntryView.propTypes = {
  closemodal: PropTypes.func,
  handleChange: PropTypes,
  toggle: PropTypes.func,
  modal: PropTypes.string,
  classname: PropTypes.string,
  title: PropTypes.string,
  body: PropTypes.string
};
export default EditEntryView;