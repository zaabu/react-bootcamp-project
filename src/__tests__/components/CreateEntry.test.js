import React from "react";
import { shallow, configure } from "enzyme";
import configureMockStore from "redux-mock-store";
import { CreateEntryModalTest } from "../../components/entries/createEntry";
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });

describe("SignupUser component", () => {
  const props = {
    handleUpdate: jest.fn(),
    handleChange: jest.fn(),
    name: "",
    type: ""
  };
  const getEvent = (name = "", value = "") => ({
    target: {
      name,
      value
    }
  });
  const mockStore = configureMockStore();
  let wrapper;

  beforeEach(() => {
    const store = mockStore({});
    wrapper = shallow(<CreateEntryModalTest {...props} store={store} />);
  });
  it("component should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
  // it("should set state when handle change is called", () => {
  //   wrapper.instance().handleChange(getEvent("name", "johnson"));
  //   expect(wrapper.state().name).toBeUndefined;
  // });
});
