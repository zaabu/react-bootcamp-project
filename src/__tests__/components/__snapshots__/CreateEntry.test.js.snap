// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignupUser component component should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CreateDiaryEntryComponent
    handleChange={[MockFunction]}
    handleUpdate={[MockFunction]}
    name=""
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    type=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <CreateEntryView
        body=""
        closemodal={[Function]}
        handleChange={[Function]}
        modal={false}
        title=""
        toggle={[Function]}
      />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "body": "",
        "classname": undefined,
        "closemodal": [Function],
        "handleChange": [Function],
        "modal": false,
        "title": "",
        "toggle": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <CreateEntryView
          body=""
          closemodal={[Function]}
          handleChange={[Function]}
          modal={false}
          title=""
          toggle={[Function]}
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "body": "",
          "classname": undefined,
          "closemodal": [Function],
          "handleChange": [Function],
          "modal": false,
          "title": "",
          "toggle": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
